php_phongo.c

```diff
@@ -3369,16 +3372,12 @@ static void php_phongo_pclient_destroy(php_phongo_pclient_t* pclient)
 	 * free_object handler or RSHUTDOWN, but there the application is capable of
 	 * freeing its Manager and its client before forking. */
 	if (pclient->created_by_pid == getpid()) {
-		/* Single-threaded clients may run commands (e.g. endSessions) from
-		 * mongoc_client_destroy, so disable APM to ensure an event is not
-		 * dispatched while destroying the Manager and its client. This means
-		 * that certain shutdown commands cannot be observed unless APM is
-		 * redesigned to not reference a client (see: PHPC-1666).
-		 *
-		 * Note: this is only relevant for request-scoped clients. APM
-		 * subscribers will no longer exist when persistent clients are
-		 * destroyed in GSHUTDOWN. */
-		mongoc_client_set_apm_callbacks(pclient->client, NULL, NULL);
+		/* If we are in request shutdown, disable APM to avoid dispatching more
+		 * events. This means that certain events (e.g. TopologyClosedEvent,
+		 * command monitoring for endSessions) may not be observed. */
+		if (EG(flags) & EG_FLAGS_IN_SHUTDOWN) {
```

Comment:
@sgolemon: If I may pick your brain for a moment: Is this something we can reasonably rely on going forward? It's hard to tell what's public and private in the Zend headers, but this seemed pretty accessible.

Reply:
Per slack. This seems to be the best you'll get for now. :(

Will followup with php-internals about this though.